### PR TITLE
Makefile: remove the unit test function timeout check temporarily

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Problem Summary:

check_dev is not stable enough.

In the past a timeout for each funcion is added to prevent single function from running too long.
However, it's hardware dependented and the CI workload is quite heavy, the time cost for a single function is not stable enough.
That has become a hinder for us to merge PRs, so it's better to disable the check temporarily.

### What is changed and how it works?

What's Changed:

Do not check function timeout in the last step in CI

How it Works:

Remove the check code.

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -113,9 +113,7 @@ ifeq ("$(TRAVIS_COVERAGE)", "1")
 else
 	@echo "Running in native mode."
 	@export log_level=info; export TZ='Asia/Shanghai'; \
-	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' $(EXTRA_TEST_ARGS) -v -cover $(PACKAGES_WITHOUT_BR) -check.p true > gotest.log || { $(FAILPOINT_DISABLE); cat 'gotest.log'; exit 1; }
-	@echo "timeout-check"
-	grep 'PASS:' gotest.log | go run tools/check/check-timeout.go || { $(FAILPOINT_DISABLE); exit 1; }
+	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' $(EXTRA_TEST_ARGS) -v -cover $(PACKAGES_WITHOUT_BR) -check.p true || { $(FAILPOINT_DISABLE); exit 1; }
```
Comment (tisonkun) [inline Makefile:116]: do we still need `-v`? Also IIRC ci pipeline now uses `cat gotest.log` for dumping logs.

**Review by dragonly — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (tiancaiamao) [reply Makefile]: OK, I remove the `-v` flag and still keep redirecting the stdout to `gotest.log` to avoid CI pipeline modification.

**Review by tisonkun — APPROVED** [review]
Thanks for submitting this PR. Looks good to me.

**Review by tangenta — APPROVED** [review]

**Pull request merged**